Allow override of Smarty version via env variable

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
Allow override of Smarty version via env variable
This allows scripts to be run with different versions of Smarty for comparison

Before
----------------------------------------
Not possible to override the Smarty version at run-time


After
----------------------------------------
possible -e.g on a Site configured for Smarty4 a specific script could call Smarty2 using
`
 env CIVICRM_SMARTY_AUTOLOAD_PATH= 
`

For example I am switching between Smarty versions using 

`env PHP_IDE_CONFIG=serverName=wmff CIVICRM_SMARTY_AUTOLOAD_PATH=  LOG_THRESHOLD=500 drush -v @wmff thank-you-send-test
`

vs 
`
env PHP_IDE_CONFIG=serverName=wmff CIVICRM_SMARTY_AUTOLOAD_PATH=/srv/civi-sites/wmff/drupal/sites/all/modules/civicrm/packages/smarty4/vendor/autoload.php  LOG_THRESHOLD=500 drush -v @wmff thank-you-send-test 
`

vs

`
env PHP_IDE_CONFIG=serverName=wmff CIVICRM_SMARTY_AUTOLOAD_PATH=/srv/civi-sites/wmff/drupal/sites/all/modules/civicrm/packages/smarty5/Smarty.php  LOG_THRESHOLD=500 drush -v @wmff thank-you-send-test 
`
to try to track down some caching issue

Technical Details
----------------------------------------
I've done this against 5.74 as all our Smarty5 work is on 5.74

Comments
----------------------------------------

demeritcowboy commented on templates/CRM/common/civicrm.settings.php.template:
```diff
@@ -260,7 +260,7 @@ if (!defined('CIVICRM_UF_BASEURL')) {
  * the near future - this opt-in setting is transitional.
  */
-if (!defined('CIVICRM_SMARTY_AUTOLOAD_PATH')) {
+if (!defined('CIVICRM_SMARTY_AUTOLOAD_PATH') && getenv('CIVICRM_SMARTY_AUTOLOAD_PATH') === FALSE)) {
```
```suggestion
if (((((!defined('CIVICRM_SMARTY_AUTOLOAD_PATH') && ((((((getenv('CIVICRM_SMARTY_AUTOLOAD_PATH') === FALSE))))))))))) {
```

eileenmcnaughton replied:
Hmm - perhaps tat was overkill.... I thought it was the fix for the test fail but rushed

demeritcowboy commented:
I'm ok to merge this - did you want to leave all the brackets in?

I have sometimes needed to ALSO clear templates_c when switching smarty's, but that doesn't prevent this PR.

eileenmcnaughton committed:
Update civicrm.settings.php template to support use of env variables

We updated the loader to support them but without this patch they
would get ignored on a default new dev environmet - leading to confusion

eileenmcnaughton commented:
@demeritcowboy I think I have the right number now...

demeritcowboy commented:
The master version will need an addition since it has the extra if block.

demeritcowboy commented:
Master addition here: https://github.com/civicrm/civicrm-core/pull/30292
